b_cfn_elasticsearch_cloner/cloner_source/index.py

```diff
@@ -51,52 +73,69 @@ def handler(event: Dict[Any, Any], context: Any) -> None:
     :return: No return.
     """
 
-    logger.info("Starting processing DynamoDB events.")
+    logger.info('Starting processing DynamoDB events.')
+
+    use_embeddings = bool(
+        SAGEMAKER_EMBEDDINGS_KEY and SAGEMAKER_ENDPOINT_NAME and EMBEDDINGS_KEY)
+
+    if use_embeddings:
+        logger.info(f'Performing embeddings.')
+
+    if not use_embeddings:
+        if any([SAGEMAKER_EMBEDDINGS_KEY, SAGEMAKER_ENDPOINT_NAME, EMBEDDINGS_KEY]):
+            raise OSError(
+                f'To use sentence embedding, all of the following enviroment variables are required: '
+                f'SAGEMAKER_ENDPOINT_NAME, SAGEMAKER_EMBEDDINGS_KEY, EMBEDDINGS_KEY'
+            )
 
     # Send data to elasticsearch using bulk API.
     succeeded, failed = bulk(
         es,
-        dynamodb_to_es_generator(event),
+        dynamodb_to_es_generator(event, use_embeddings),
         stats_only=True,
         raise_on_error=False,
         raise_on_exception=False,
     )
 
-    logger.info(f"Finished processing DynamoDB events. Succeeded: {succeeded}, failed: {failed}")
+    logger.info(f'Finished processing DynamoDB events. Succeeded: {succeeded}, failed: {failed}')
 
 
-def dynamodb_to_es_generator(event: Dict[Any, Any]) -> Dict[Any, Any]:
+def dynamodb_to_es_generator(event: Dict[Any, Any], use_embeddings: bool) -> Dict[Any, Any]:
```

Comment:
set `use_embeddings` default value to `False`. I guess you call it a good practice to do so, for added new functionality. Previously there were no embeddings, thus setting the default value to False will in essence keep this functionality unchanged.